### PR TITLE
Sort on incident score

### PR DESCRIPTION
**Pull request by frenchy64**
**Epic** https://github.com/advthreat/iroh/issues/7341
Close https://github.com/advthreat/iroh/issues/7347
Related https://github.com/advthreat/iroh/issues/7344

Adds the ability to sort on incident score. New sorting fields are allowed of the form `scores.X`, `scores.Y` on incidents, when a configuration `ctia.http.incident.sortable-score-types=X,Y` is provided.

These fields can be combined as usual for more complex (nested) sorts.

<a name="qa">[§](#qa)</a> QA
============================

No QA is needed.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: Sort on incident score
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

**Review by ereteog — COMMENTED** [review]
It looks good.
- can you move to the config the score types, please? I think that the ttp order should also be moved to the conf.
- I would like to discuss the sort schema (see my comment)

File: src/ctia/entity/incident.clj [hunk 1]
```diff
@@ -206,36 +209,45 @@
...
+                 :remap-default 0}}
+      ;; :scores.asset, :scores.ttp
+      ;; Sort by maximum score of a particular type
+      (into (map (fn [score-type]
+                   {(keyword (str "scores." score-type))
+                    {:op :sort-by-list-max
+                     :field-name "scores"
+                     :max-entry "score"
+                     :filter-entry {"type" score-type}}}))
+            ["asset" "ttp"])))
```
[the remark this answers is not in the document]
Comment (frenchy64) [reply src/ctia/entity/incident.clj]: Moved to config: `ctia.http.incident.sortable-score-types`.

File: src/ctia/schemas/core.clj [hunk 2]
```diff
@@ -346,7 +346,11 @@
     #(= :remap (:op %)) {:op (s/eq :remap)
                          (s/optional-key :field-name) (s/cond-pre s/Keyword s/Str)
                          :remappings {s/Str s/Num}
-                         :remap-default s/Num}))
+                         :remap-default s/Num}
+    #(= :sort-by-list-max (:op %)) {:op (s/eq :sort-by-list-max)
+                                    :field-name (s/cond-pre s/Keyword s/Str)
+                                    :max-entry s/Str
+                                    (s/optional-key :filter-entry) {s/Str s/Str}}))
```
[the remark this answers is not in the document]
Comment (frenchy64) [reply src/ctia/schemas/core.clj]: I implemented both of your suggestions.

**Review by ereteog — APPROVED** [review]
LGTM (though it's blocked by the fulltext test)

**Pull request merged**